src/detect-jsx.js

```diff
@@ -0,0 +1,19 @@
+const { parse } = require("@babel/parser");
+const traverse = require("../babel-traverse/lib/index.js").default;
+
+const { parseOptions } = require("./convert.js");
+
+const detectJsx = tsCode => {
+  let jsx = false;
+  const ast = parse(tsCode, parseOptions);
```

Comment:
It would be nice to avoid extra calls to `parse` but that could be a separate PR since changes will need to be made elsewhere.